uswds/uswds
# USWDS - Date Picker: Updated toggleCalendar function to account for aria-disabled attribute + readonly

# Summary

**Updated toggleCalendar function to account for `aria-disabled attribute + readonly`**. The new code added prevents users from selecting a date while allowing the button to remain discoverable, which is required for accessibility. It also adds the `readonly` which effectively replicates disabled behaviors without the accessibility concerns.

## Breaking change

This is not a breaking change.

## Related issue

Closes #5982 

## Related PR
[Changelog PR](https://github.com/uswds/uswds-site/pull/2794)

## Preview link

Preview link:
[Date Picker](https://federalist-3b6ba08e-0df4-44c9-ac73-6fc193b0e19c.sites.pages.cloud.gov/preview/uswds/uswds/rc-prevent-calendar-form-opening-aria-disabled/?path=/story/components-form-inputs-date-picker--range-date)

## Problem statement

The date picker does not fully comply with accessibility requirements. Users who rely on assistive technologies may experience frustration or confusion while attempting to interact with the date picker, and users who follow recommended disabled guidance may find it difficult to do so.

## Solution

This PR adds a check for the `aria-disabled` attribute to the toggleCalendar() function, ensuring that the date picker remains discoverable while preventing users from selecting a date when the element is disabled. It also adds the `readonly` attribute when `aria-disabled` is used to help the date picker be discoverable by screenreaders. This change aligns with accessibility requirements and simplifies the process for users to follow recommended disabled guidance.

## Testing and review

1. Under 'Date Picker - Range Date' in Storybook, change "Disabled State" to aria-disabled. Refresh page to make change.
2. Ensure calendar form does not open when calendar icon is clicked
3. Use a screen reader to test that the toggle button is announced as disabled.


Before opening this PR, make sure you’ve done whichever of these applies to you:
